noir/noir-repo/tooling/noirc_abi/src/lib.rs

```diff
@@ -513,30 +516,81 @@ fn decode_string_value(field_elements: &[FieldElement]) -> String {
     final_string.to_owned()
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ContractEvent {
-    /// Event name
-    name: String,
-    /// The fully qualified path to the event definition
-    path: String,
-
-    /// Fields of the event
-    #[serde(
-        serialize_with = "serialization::serialize_struct_fields",
-        deserialize_with = "serialization::deserialize_struct_fields"
-    )]
-    fields: Vec<(String, AbiType)>,
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "kind", rename_all = "lowercase")]
+pub enum AbiValue {
+    Field {
+        value: FieldElement,
+    },
+    Integer {
+        sign: bool,
+        value: String,
+    },
```

Comment:
We can serialize this to an `i128` (large enough to hold any Noir integer value and then serialize that to a string.

I don't think you're going to be getting the value you want written to the ABI currently due to how we handle negative integers internally.

Reply:
Done

Reply:
Tried this but we really need to export full fields too, which seems to be inside the `HirLiteral::Integer` enum, so we can't downcast to `i128`. Kept the `to_string`, but changed the ts side to use strings and not bigints.